ddtrace/tracer/remote_config.go

```diff
@@ -220,7 +219,8 @@ func (t *tracer) onRemoteConfigUpdate(u remoteconfig.ProductUpdate) map[string]s
 		if updated {
 			telemConfigs = append(telemConfigs, t.config.traceSampleRate.toTelemetry())
 		}
-		updated = t.config.traceSampleRules.handleRC(convertRemoteSamplingRules(c.LibConfig.SamplingRules))
+		// TODO: report raw sampling rules to telemetry
```

Comment:
can you explain this TODO? I think/thought the remote sampling rules reporting was already implemented. Maybe you mean something else?

Reply:
@lucaspimentel told me that ideally telemetry should report the original JSON received through RC. Right now we don't do that, we report what we were able to parse into a struct, hence the `convertRemoteSamplingRules` method.

Reply:
I see. Reporting the original json is not possible by its literal meaning in this case because the remote rules does not take the same json format due to RC's security restrictions - we cannot use arbitrary maps which is required in the original form of tags.

In this case, we should just take the gist of the "original json", meaning the same format of what original json would be. We can remove this TODO.

Reply:
Removed. Merging when CI is green. Thanks @yuanyuanzhao3 for your switf review!

Reply:
The TODO comment was correct in the sense that the Go tracer should (someday) revisit how it handles configuration telemetry. Tracers should send to telemetry the original (raw) config strings as they were before any marshaling/parsing/deserialization happens. This is true for all settings, whether they are interpreted are strings, numbers, Boolean, or complex objects like the sampling rules.

For example, if a user sets `DD_TRACE_SAMPLING_RULES` to a string that is invalid JSON, we still want to send the original string to telemetry, and populate the `error` field to indicate that the string could not be parsed.

> the remote rules does not take the same json format

The different json schemas shouldn't be an issue since the tracing libraries should report the original strings it receives from all config sources _separately_, each one with a different value in the `origin` field (`env_var`, `remote_config`, etc). See the [Instrumentation Telemetry API Docs](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/SchemaDocumentation/Schemas/conf_key_value.md#properties) for details.

> In this case, we should just take the gist of the "original json", meaning the same format of what original json would be.

The tracing library should send to config telemetry the exact configuration string it receives without any change, regardless if its origin is env vars or from remote config.